src/protocols/secure_channel/PASESession.cpp

```diff
@@ -482,38 +552,44 @@ CHIP_ERROR PASESession::HandlePBKDFParamResponse(const System::PacketBufferHandl
 exit:
     if (err != CHIP_NO_ERROR)
     {
-        SendErrorMsg(Spake2pErrorType::kUnexpected);
+        SendStatusReport(kProtocolCodeInvalidParam);
     }
     return err;
 }
 
 CHIP_ERROR PASESession::SendMsg1()
 {
+    size_t data_len                = EstimateTLVStructOverhead(kMAX_Point_Length, 1);
+    System::PacketBufferHandle msg = System::PacketBufferHandle::New(data_len);
+    VerifyOrReturnError(!msg.IsNull(), CHIP_ERROR_NO_MEMORY);
+
+    System::PacketBufferTLVWriter tlvWriter;
+    tlvWriter.Init(std::move(msg));
+
+    TLV::TLVType outerContainerType = TLV::kTLVType_NotSpecified;
+    ReturnErrorOnFailure(tlvWriter.StartContainer(TLV::AnonymousTag, TLV::kTLVType_Structure, outerContainerType));
+
     uint8_t X[kMAX_Point_Length];
     size_t X_len = sizeof(X);
 
     ReturnErrorOnFailure(
         mSpake2p.BeginProver(nullptr, 0, nullptr, 0, mPASEVerifier.mW0, kSpake2p_WS_Length, mPASEVerifier.mL, kSpake2p_WS_Length));
-
     ReturnErrorOnFailure(mSpake2p.ComputeRoundOne(NULL, 0, X, &X_len));
+    ReturnErrorOnFailure(tlvWriter.Put(TLV::ContextTag(1), ByteSpan(X, X_len)));
```

Comment:
Suggest an empty line above this one.

Also, any usage of `ByteSpan(x, sizeof(x))` where `x` is fixed length array can be replaced with just `ByteSpan(x)`  and avoid the sizeof argument

Reply:
Added the empty line.

The length of X (X_Len) could potentially be updated in `mSpake2p.ComputeRoundOne`. Just for accounting for that condition, I had to pass X_Len as the second parameter.

Reply:
Hmmm. How can this length actually change, from a logical standpoint?

Reply:
I believe the intent of the API is that the output buffer could be longer than needed. In this scenario, we are providing the exact sized buffer. Maybe the code can be changed to assert that returned buffer length matches what we provided.

Reply:
Added the check for length, and removed the length from call to ByteSpan() ctor.